game/src/core/components/FileSystemUtil.java

```diff
@@ -0,0 +1,219 @@
+   */
+  private static boolean isStartedInJUnitTest() {
+    for (StackTraceElement element : Thread.currentThread().getStackTrace()) {
+      if (element.getClassName().startsWith("org.junit.")) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  /**
+   * This method tries to recursively visit resources using the test resources directory. It checks
+   * if the method is called in a JUnit test and throws an exception if not. Then it walks the file
+   * tree starting from the specified path, using the provided file visitor.
+   *
+   * @param path the path to the resources
+   * @param visitor the file visitor to use
+   * @throws Exception if the method is not called in a JUnit test
+   */
+  private static void visitJUnitResourcesViaWalkFileTree(
+      final String path, final SimpleFileVisitor<Path> visitor) throws Exception {
+    if (!isStartedInJUnitTest()) {
+      throw new IllegalStateException(
+          "FileSystemUtil.visitResourcesViaJUnit can only be used in JUnit tests");
+    }
+    Files.walkFileTree(
+        Paths.get(
+            Objects.requireNonNull(Thread.currentThread().getContextClassLoader().getResource(path))
+                .toURI()
+                .normalize()),
+        visitor);
+  }
+
+  /**
+   * This method tries to recursively visit resources using the context class loader. If it fails,
+   * an exception is thrown. Then it walks the file tree starting from the specified path, using the
+   * provided file visitor.
+   *
+   * @param path the path to the resources
+   * @param visitor the file visitor to use
+   * @throws Exception if the method is not called in a JUnit test
+   */
+  private static void visitResourcesViaGetResourceAsStream(
+      final String path, final SimpleFileVisitor<Path> visitor) throws Exception {
+    final ClassLoader loader = Thread.currentThread().getContextClassLoader();
+    try (final InputStream is = loader.getResourceAsStream(path)) {
+      assert is != null;
+      try (final InputStreamReader isr = new InputStreamReader(is, StandardCharsets.UTF_8);
+          final BufferedReader br = new BufferedReader(isr)) {
+        List<String> list = br.lines().toList();
+        if (list.isEmpty()) {
+          throw new IOException("Empty: " + path);
+        }
+        list.stream()
+            .map(l -> path + "/" + l)
+            .forEach(
+                p -> {
+                  try {
+                    final Path path2 =
+                        Paths.get(
+                            Objects.requireNonNull(loader.getResource(p.replace("//", "/")))
+                                .toURI());
+                    visitor.visitFile(path2, null);
+                    if (Files.isDirectory(path2)) {
+                      visitResourcesViaGetResourceAsStream(p, visitor);
+                    }
+                  } catch (Exception e) {
+                    throw new RuntimeException(e);
+                  }
```

Comment:
Was hast Du nur immer mit diesem Konstrukt, das taucht in diesem PR mehrfach auf?! Was ist der Zweck davon? Wenn Du ne neue RTE wirfst, kannst Du doch gleich auf das Catchen verzichten und einfach die ursprüngliche Exception hochreichen?!